File: Laba2/Var31.py
import math
pi=3.14
x = int(input("Введите x:"))
a = int(input("Введите a:"))
z = str(input("Введите букву,которую xотите вычислить"))
F = 0
G = 0
Y = 0
if z == "G":
    G = 5 * (10 * a ** 2 - 11 * a * x + x ** 2) / (24 * a ** 2 - 49 * a * x + 15 * x ** 2)
    print("G =", G)
elif z == "Y":
    Y = math.atan(30 * a ** 2 + 37 * a * x - 4 * x ** 2) * -1
    if Y:
        print("Y =", Y)

elif z == "F":
    F = math.sinh(3 * a ** 2 + 7 * a * x + 4 * x ** 2)
    if F:
        print("F =", F)
    else: print("Не существует значения")

File: Laba2/Var31.c
#include <stdio.h>
#include <math.h>

int main() {
    double G, F, Y, x, a;
    char r;

    printf("%s", "Введите x: ");
    scanf("%lf", &x);
    getchar();

    printf("%s", "Введите a: ");
    scanf("%lf", &a);
    getchar();

    printf("Введите 1 для вызова G, введите 2 для вызова F, введите 3 для вызова Y  \n");
    scanf("%c", &r);

    switch (r) {
        case 1:
            if ((G <= 0) || (G >= 0)) {
                G = ((5 * (10 * pow(a, 2) - 11 * a * x + pow(x, 2))) / (24 * pow(a, 2) - 49 * a * x + 4 * pow(x, 2)));
                printf("%lf \n", G);
            } else printf("%s\n", "При данном значение G выполнение функции невозможно");
            break;
        case 2:
            F = sinh(3 * pow(a, 2) + 7 * a * x + 4 * pow(x, 2));
            printf("F = %lf \n", F);
            break;
        case 3 :
            if ((-1 <Y) ||  (Y < 1)) {
                Y = tanh(30 * pow(a, 2) + 37 * a * x - 4 * pow(x, 2));
            } else printf("%s\n", "При данном значение Y выполнение функции невозможно");
            printf("Y = %lf \n", Y);
            break;
        default:
            printf("%s\n", "Для выполнения действия напишите 1,2 или 3");
    }

    return 0;
